analyzer/quality-analyzer/src/main/java/eu/fasten/analyzer/qualityanalyzer/QualityAnalyzerPlugin.java

```diff
@@ -0,0 +1,173 @@
+package eu.fasten.analyzer.qualityanalyzer;
+
+import eu.fasten.core.plugins.KafkaPlugin;
+import eu.fasten.core.plugins.DBConnector;
+import eu.fasten.core.data.metadatadb.MetadataDao;
+
+import org.jooq.exception.DataAccessException;
+import org.jooq.impl.DSL;
+import org.json.JSONObject;
+import org.pf4j.Extension;
+import org.pf4j.Plugin;
+import org.pf4j.PluginWrapper;
+
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.List;
+
+import org.jooq.DSLContext;
+
+
+public class QualityAnalyzerPlugin extends Plugin {
+
+    public QualityAnalyzerPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class QualityAnalyzer implements KafkaPlugin, DBConnector {
+
+        private String consumerTopic = "fasten.RapidPlugin.out";
+        private static DSLContext dslContext;
+        private final Logger logger = LoggerFactory.getLogger(QualityAnalyzer.class.getName());
+        private Throwable pluginError = null;
+        private String product = null;
+        private String artifact = null;
+        private String group = null;
+        private String version = null;
+        private JSONObject metrics = null;
+        private boolean restartTransaction = false;
+        private final int transactionRestartLimit = 3;
```

Comment:
`transactionRestartLimit` is now a part of Constants class in `core/data/Constants`, so please reference it from there